src/pages/index.js

```diff
@@ -43,61 +80,152 @@ const profileEditPopup = new PopupWithForm(
-/* -------------------------------------------------------------------------- */
+/* ---------------------------Confirm Delete Popup-------------------------- */
 
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: (item) => {
-      console.log(item);
-      const cardEl = renderCard(item);
-      cardSection.addItem(cardEl);
-    },
-  },
-  ".cards__list"
+const confirmDeletePopup = new PopupWithConfirmation(
+  "#confirm-image-delete-popup",
+  handleDeleteConfirmSubmit
 );
-cardSection.renderItems();
+confirmDeletePopup.setEventListeners();
 
-const userInfoNew = new UserInfo(".profile__name", ".profile__description");
+/* ------------------------Change Profile Avatar Popup---------------------- */
+
+const changeProfileAvatarPopUp = new PopupWithForm(
+  "#change-profile-avatar-popup",
+  handleChangeProfileAvatarFormSubmit
+);
+changeProfileAvatarPopUp.setEventListeners();
+
+const userInfoNew = new UserInfo(
+  profileName,
+  profileDescription,
+  profileAvatar
+);
+
+api
+  .fetchUserInfo()
+  .then((data) => (userData = data))
+  .then((data) => {
+    userInfoNew.setUserInfo(userData.name, userData.about);
+    userInfoNew.setUserAvatar(userData.avatar);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
 
 /* -------------------------------------------------------------------------- */
 /*                                  Functions                                 */
 /* -------------------------------------------------------------------------- */
 
 function renderCard(data) {
-  const card = new Card(data, "#card-template", handleImageClick);
+  const card = new Card(
+    data,
+    "#card-template",
+    handleImageClick,
+    handleDeleteConfirmSubmit,
+    handleCardLike
+  );
   return card.getView();
 }
 
 function handleImageClick(data) {
-  console.log(data);
   imagePopUp.open(data);
 }
 
 function handleEditProfileFormSubmit(data) {
-  userInfoNew.setUserInfo(data);
-  profileEditPopup.close();
+  profileEditPopup.setLoading(true, "Saving...");
+  api
+    .editUserInfo(data)
+    .then(() => {
+      userInfoNew.setUserInfo(data);
+      profileEditPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => profileEditPopup.setLoading(false, "Save"));
+}
+
+function handleChangeProfileAvatarFormSubmit(data) {
+  changeProfileAvatarPopUp.setLoading(true, "Saving...");
+  api
+    .editUserAvatar(data.link)
+    .then(() => {
+      userInfoNew.setUserAvatar(data.link);
+      changeProfileAvatarPopUp.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => changeProfileAvatarPopUp.setLoading(false, "Save"));
 }
 
 function handleAddCardFormSubmit(data) {
-  const cardValue = renderCard(data);
-  cardSection.addItem(cardValue);
-  addCardPopUp.close();
+  addCardPopUp.setLoading(true, "Saving...");
+  api
+    .addCard(data)
+    .then((res) => card)
```

Comment:
Needs correction: there is an error in the console when you add a card `index.js:187 ReferenceError: card is not defined
    at index.js:179:20`.

You should use one `then` block

```js
 .addCard(data)
.then((card) => 
```